roles/install-heimdall/tasks/main.yml

```diff
@@ -1,33 +1,24 @@
 ---
 # tasks file for heimdall
 
-- name: Clone the Heimdall
-  git:
-    repo: "https://github.com/maticnetwork/heimdall.git"
-    dest: "{{ ansible_env.HOME }}/heimdall"
-    force: yes
-    version: "{{ heimdall_branch }}"
+- name: install heimdall pacakges and heimdall profile
+  ansible.builtin.shell:
+    cmd: curl -L https://raw.githubusercontent.com/maticnetwork/install/main/heimdall.sh | bash -s -- {{ heimdall_version }} {{ network }} {{ node_type }}
 
-- name: Build heimdall
-  command: chdir={{ ansible_env.HOME }}/heimdall make install network={{ heimdall_network }}
+- name: Setup Mainnet Heimdall Seeds
+  ansible.builtin.command:
+    cmd: sudo sed -i 's|^seeds =.*|seeds = "d3a8990f61bb3657da1664fe437d4993c4599a7e@3.211.248.31:26656,d3d7d397339c9126235dfab11bf925e269776f4f@3.212.183.151:26656,68254d33685fad151e45bfe1ed33d538ba6ec8cb@3.93.224.197:26656,d26c54ebbf274896f12977bb13d83ac1237a8226@184.73.124.158:26656,f4f605d60b8ffaaf15240564e58a81103510631c@159.203.9.164:26656,4fb1bc820088764a564d4f66bba1963d47d82329@44.232.55.71:26656,2eadba4be3ce47ac8db0a3538cb923b57b41c927@35.199.4.13:26656,25f5f65a09c56e9f1d2d90618aa70cd358aa68da@35.230.116.151:26656,3b23b20017a6f348d329c102ddc0088f0a10a444@35.221.13.28:26656"|g' /var/lib/heimdall/config/config.toml
+  when: network == "mainnet"
 
-- name: Creating symlink for heimdalld
-  file:
-    src: "{{ ansible_env.HOME }}/go/bin/heimdalld"
-    dest: "/usr/bin/heimdalld"
-    state: link
-  become: true
+- name: Setup Mumbai Heimdall Seeds
+  ansible.builtin.command:
+    cmd: sudo sed -i 's|^seeds =.*|seeds = "b18bbe1f3d8576f4b73d9b18976e71c65e839149@34.226.134.117:26656,4cd60c1d76e44b05f7dfd8bab3f447b119e87042@54.147.31.250:26656,7a6c7b5d25b13ce3448b047dbebeb1a19cc2e092@18.213.200.99:26656"|g' /var/lib/heimdall/config/config.toml
+  when: network == "mumbai"
+
+- name: start Heimdall service
+  ansible.builtin.shell:
+    cmd: |
+       sudo systemctl daemon-reload
+       sudo systemctl start heimdalld.service
```

Comment:
There is an ansible module that you can use to manage systemd as well, so you don't have to use shell/cmd. https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_module.html